dev/devicelab/lib/framework/utils.dart

```diff
@@ -0,0 +1,403 @@
+// Copyright (c) 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:stack_trace/stack_trace.dart';
+
+/// Virtual current working directory, which affect functions, such as [exec].
+String cwd = Directory.current.path;
+
+List<ProcessInfo> _runningProcesses = <ProcessInfo>[];
+
+class ProcessInfo {
+  ProcessInfo(this.command, this.process);
+
+  final DateTime startTime = new DateTime.now();
+  final String command;
+  final Process process;
+
+  @override
+  String toString() {
+    return '''
+  command : $command
+  started : $startTime
+  pid     : ${process.pid}
+'''
+        .trim();
+  }
+}
+
+/// Result of a health check for a specific parameter.
+class HealthCheckResult {
+  HealthCheckResult.success([this.details]) : succeeded = true;
+  HealthCheckResult.failure(this.details) : succeeded = false;
+  HealthCheckResult.error(dynamic error, dynamic stackTrace)
+      : succeeded = false,
+        details = 'ERROR: $error\n${stackTrace ?? ''}';
+
+  final bool succeeded;
+  final String details;
+
+  @override
+  String toString() {
+    StringBuffer buf = new StringBuffer(succeeded ? 'succeeded' : 'failed');
+    if (details != null && details.trim().isNotEmpty) {
+      buf.writeln();
+      // Indent details by 4 spaces
+      for (String line in details.trim().split('\n')) {
+        buf.writeln('    $line');
+      }
+    }
+    return '$buf';
+  }
+}
+
+class BuildFailedError extends Error {
+  BuildFailedError(this.message);
+
+  final String message;
+
+  @override
+  String toString() => message;
+}
+
+void fail(String message) {
+  throw new BuildFailedError(message);
+}
+
+void rm(FileSystemEntity entity) {
+  if (entity.existsSync()) entity.deleteSync();
```

Comment:
`dartfmt` did this. I prefer next line too. Fixed. I also prefer using braces here to avoid stuff like [this](http://www.dwheeler.com/essays/apple-goto-fail.html) but alas, Flutter style guide says no braces.